azext_iot/product/command_tests.py

```diff
@@ -142,14 +146,125 @@ def initialize_workspace(cmd, product_name, working_folder="PnPCert"):
+    attestation_type=None,
+    certificate_path=None,
+    endorsement_key=None,
+    badge_type=BadgeType.IotDevice.value,
+    models=None,
+    provisioning=False
+):
     # call to POST /deviceTests
-    return True
+    if (attestation_type == AttestationType.x509.value and not certificate_path):
+        raise CLIError('If attestation type is x509, certificate path is required')
+    if (attestation_type == AttestationType.tpm.value and not endorsement_key):
+        raise CLIError('If attestaion type is tpm, endorsement key is required')
+    if (badge_type == BadgeType.Pnp.value and not models):
+        raise CLIError('If badge type is Pnp, models is required')
+    if not any(
+        [
+            configuration_file,
+            all([
+                device_type,
+                product_id,
+                attestation_type,
+                badge_type
+            ])
+        ]
+    ):
+        raise CLIError('If configuration file is not specified, attestation and device definition parameters must be specified')
+    test_configuration = _create_from_file(configuration_file) if configuration_file else _build_test_configuration(
+        product_id,
+        device_type,
+        attestation_type,
+        certificate_path,
+        endorsement_key,
+        badge_type,
+        models)
+    return get_sdk(cmd).create_device_test(provisioning, body=test_configuration)
+
+
+def _build_test_configuration(product_id, device_type, attestation_type, certificate_path, endorsement_key, badge_type, models):
+    config = {
+        'validationType': 'Certification',
+        'productId': product_id,
+        'deviceType': device_type,
+        'provisioningConfiguration': {
+            'type': attestation_type
+        },
+        'certificationBadgeConfigurations': [
+            {
+                'type': badge_type
+            }
+        ]
+    }
+    if (attestation_type == AttestationType.symmetricKey.value):
+        config['provisioningConfiguration']['symmetricKeyEnrollmentInformation'] = {}
+    elif (attestation_type == AttestationType.tpm.value):
+        config['provisioningConfiguration']['tpmEnrollmentInformation'] = {
+            'endorsementKey': endorsement_key
+        }
+    elif (attestation_type == AttestationType.x509.value):
+        config['provisioningConfiguration']['x509EnrollmentInformation'] = {
+            'base64EncodedX509Certificate': _read_certificate_from_file(certificate_path)
+        }
+    if (badge_type == BadgeType.Pnp.value and models):
+        models_array = _process_models_directory(models)
+        config['certificationBadgeConfigurations'][0]['digitalTwinModelDefinitions'] = models_array
+
+    return config
+
+
+def _read_certificate_from_file(certificate_path):
+    with open(file=certificate_path, mode='rb') as f:
+        data = f.read()
+
+        from base64 import encodestring
+        return encodestring(data)
+
+
+def _process_models_directory(from_directory):
```

Comment:
I think we may want to pull this into PnP at some point as well for bulk model validation - not necessarily now but we should probably put this in a utils class eventually

Reply:
Yeah, I was debating on moving it now or later. I didn't want to directly reference the once in digitaltwins.